docs/src/references/c8y-configuration-management.md

```diff
@@ -162,4 +164,179 @@ Note that:
+
+**Device-to-Cloud Behaviour:**
+  * at some point a config retrieval for type `bar.conf` for `child1` arrives at C8Y config plugin
+  * C8Y config plugin: subscribes to `configs/bar.conf` and gets immediately the config file content from the retained message
+  * C8Y config plugin: sends the config file to C8Y<br/><br/>
+    NOTE: The responsibility to assure that the latest config file content is on the MQTT bus is always on the process/external device.
+When there is no retain message the plugin sends an error to the cloud on an every incoming config retrieval request.
+
+**Cloud-to-Device Behaviour:**
+  * at some point a config sent from cloud for type `bar.conf` for `child1` arrives at C8Y config plugin
+  * C8Y config plugin: publishs the config file content as MQTT retain message to `configs/bar.conf` 
+  * external device `child1`: recognizes the MQTT message on `configs/bar.conf` and processes the new config file content
+
+
+## Declaration of Capabilities for Child-Devices
+
+The `c8y_configuration_plugin` takes care to define all necessary capabilities to each cloud's child-device twin. These are:
+  - declaring _supported operations_ for configuration management: `c8y_UploadConfigFile` and `c8y_DownloadConfigFile`
+  - declaring provided _configuration types_
+
+**Declaring 'supported operations'**
+
+For each `childid` contained in the plugin's configuration, the plugin sends one MQTT message to C8Y. If one `childid` occurs more than once just one message will be sent. The message is as below:
+  - topic: `c8y/s/us/<childid>`
+  - payload: `114,c8y_UploadConfigFile,c8y_DownloadConfigFile`
+
+
+**Declaring 'provided configuration types'**
+
+For each `childid` contained in the plugin's configuration, the plugin sends one MQTT message to C8Y. Thereby all `config types` assigned to that `childid` will be combined in that single message. The message is as below:
+  - topic: `c8y/s/us/<childid>`
+  - payload: `119,<config type 1>,<config type 2>,<config type 3>,...`
+
+
+**When to send capability messages**
+
+The `c8y_configuration_plugin` sends both MQTT message from above (for _supported operations_ and _provided configuration types_), always when the plugin's configuration is (re)processed. That means:
+  - on startup of the plugin
+  - always when the plugin recognizes a configuration change and processes it's configuration again
+
+
+Note that the `c8y_configuration_plugin` does **not** create any child-device twin in the cloud. Instead the clouds child-device twins must be created upfront.
+
+
+## Notfifications for Child-Devices
+
+Notifications for child-devices behave the same way as nofifications for the thin-edge device (as described in section _Notifications_ above), but are extended with the `childid`. That is since the `type` is unique for one device, but can again occur on another child-device or the thin-edge device.
```

Comment:
By moving this section along the notification section, this will move away cross references, make things clearer and help to spot design errors.

Reply:
Meanwhile the entire section "Notfifications for Child-Devices" was removed as deprecated, see https://github.com/cstoidner/thin-edge.io/commit/9db6a56b1c45d5d9eee169b443f8a2b631cf5063